### PR TITLE
BLD: fix requirements

### PR DESCRIPTION
**Pull request by klauer**
## Description
* I had these requirements fixed locally in the first commit
* The remaining commits are here to get CI going

## Motivation and Context
* I'm not sure if this was a WIP or complete; opening this PR before I forget

### CODE REVIEW

**Comment by klauer** [comment]
OK, I went wrong somewhere on the setuptools-scm config on conda

**Comment by klauer** [comment]
Motivation section states: "  I'm not sure if this was a WIP or complete; opening this PR before I forget"

Months later, am still not sure

File: pyproject.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 build-backend = "setuptools.build_meta"
 requires = [
     "setuptools>=45",
-    "setuptools_scm[toml]>=6.2"
+    "setuptools_scm[toml]>=8.0"
```
Comment (tangkong) [inline pyproject.toml]:
```suggestion
    "setuptools_scm[toml]>=6.2"
```

File: pyproject.toml [hunk 2]
```diff
@@ -37,7 +37,7 @@ file = "LICENSE"
 "lclsspeak" = "lclsspeak.bin.main:main"
 
 [tool.setuptools_scm]
-write_to = "lclsspeak/_version.py"
+version_file = "lclsspeak/_version.py"
```
Comment (tangkong) [inline pyproject.toml]:
```suggestion
write_to = "lclsspeak/_version.py"
```

we pinned the ci-helpers to setuptools_scm < 8.0, so we use the old `write_to`.

**Review by tangkong — APPROVED** [review]
LGTM, god bless slack search history

**Pull request merged**